### PR TITLE
#6 Search Module Improvement

### PR DESCRIPTION
**Pull request by hkulekci**
- Added IndexingException and update MissingDataException comment
 - Added partial update
 - Added delete method to delete document by id
 - Added getById method to AbstractQueryService

### CODE REVIEW

File: module/Search/src/Search/Query/Service/AbstractQueryService.php [hunk 1]
```diff
@@ -23,6 +24,29 @@ class AbstractQueryService extends AbstractSearchService
...
+     *
+     * @throws SearchingException
+     *
+     * @return array
+     */
+    public function getById($docId, $version = 1)
+    {
+        try {
+            return $this->getClient()
+                        ->getIndex($this->getIndexName($version))
```
Comment (edigu) [inline module/Search/src/Search/Query/Service/AbstractQueryService.php]: Using a signature like `->getIndex ($version)` and `->getType ($version)` would more consistent with the Indexing service.

File: module/Search/src/Search/Index/Service/AbstractIndexService.php [hunk 2]
```diff
@@ -25,7 +31,54 @@ public function index(array $data, $version = 1)
...
+        try {
+            $this->getType($version)->updateDocument($document);
+        } catch (\Exception $e) {
+            throw new IndexingException('Throw exception while updating', $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * Partial Update
+     *
```
Comment (edigu) [inline module/Search/src/Search/Index/Service/AbstractIndexService.php]: Docblock comment..

**Comment by edigu** [comment]
LGTM. Just few small notes

File: module/Search/src/Search/Query/Service/AbstractQueryService.php (hunk 1 above)
Comment (hkulekci) [inline module/Search/src/Search/Query/Service/AbstractQueryService.php]: oops, thats right. I will chage.

**Comment by hkulekci** [comment]
updated!

**Comment by hkulekci** [comment]
:

**Pull request merged**